src/components/calendar-bookings/CalendarBookingsForm.tsx

```diff
@@ -0,0 +1,209 @@
+import * as React from 'react'
+import { NylasScheduling } from '@nylas/react'
+import { Collapse, Fade } from '@mui/material'
+import {
+  schedulingService,
+  submissionService,
+  OneBlinkAppsError,
+} from '@oneblink/apps'
+import clsx from 'clsx'
+import { Receipt } from '../receipts'
+import ErrorModal from './ErrorModal'
+import OnLoading from '../renderer/OnLoading'
+import CalendarBookingsPage from './CalendarBookingsContainer'
+import useCalendarBookings from './CalendarBookingsProvider'
+
+function NylasBookingForm({
+  submissionId,
+  name,
+  email,
+  configurationId,
+  sessionId,
+  onBookingConfirmed,
+  postSubmissionAction,
+}: Awaited<
+  ReturnType<typeof schedulingService.createNylasNewBookingSession>
+> & {
+  submissionId: string
+  postSubmissionAction: (
+    formSubmissionResult: submissionService.FormSubmissionResult,
+  ) => void
+}) {
+  const { setBookingError, onTimeSlotConfirmed } = useCalendarBookings()
+
+  const [
+    {
+      formSubmissionResult,
+      isRunningPostSubmissionAction,
+      postSubmissionError,
+      isConfirmingBooking,
+      confirmingBookingError,
+    },
+    setPostSubmissionState,
+  ] = React.useState<{
+    formSubmissionResult: submissionService.FormSubmissionResult | null
+    isRunningPostSubmissionAction: boolean
+    postSubmissionError: OneBlinkAppsError | null
+    isConfirmingBooking: boolean
+    confirmingBookingError: OneBlinkAppsError | null
+  }>({
+    formSubmissionResult: null,
+    isRunningPostSubmissionAction: false,
+    postSubmissionError: null,
+    isConfirmingBooking: false,
+    confirmingBookingError: null,
+  })
+  const clearPostSubmissionError = React.useCallback(() => {
+    setPostSubmissionState((currentState) => ({
+      ...currentState,
+      postSubmissionError: null,
+    }))
+  }, [])
+  const clearConfirmingBookingError = React.useCallback(() => {
+    setPostSubmissionState((currentState) => ({
+      ...currentState,
+      confirmingBookingError: null,
+    }))
+  }, [])
+
```

Comment:
This file should also include the `executePostSubmissionAction()` that is in the renderer as we want the error handling and such. We should be called `onDone()` within that function instead of `submissionService.executePostSubmissionAction()`.